tests/qat/test_instructions.py

```diff
@@ -481,3 +483,120 @@ def test_most_instructions(self):
+        if entangled_qubits is None:
+            assert mb._entangled_qubits == set(targets)
+        else:
+            assert mb._entangled_qubits == set(entangled_qubits)
+
+    def test_add_to_measure_block(self):
+        hw = get_default_echo_hardware()
+        targets = [hw.qubits[0], hw.qubits[-1]]
+        modes = [AcquireMode.INTEGRATOR, AcquireMode.SCOPE]
+        out_vars = ["c[0]", "b[1]"]
+        entangled_qubits = hw.qubits[:2]
+        mb = MeasureBlock(
+            targets[0],
+            modes[0],
+            output_variables=out_vars[:1],
+            entangled_qubits=entangled_qubits,
+        )
+        assert mb.quantum_targets == hw.qubits[:1]
+        mb.add_measurements(targets[1], modes[1], output_variables=out_vars[1])
+        entangled_qubits.append(targets[1])
+        assert mb.quantum_targets == targets
+        assert [val["mode"] for val in mb._target_dict.values()] == modes
+        assert [val["output_variable"] for val in mb._target_dict.values()] == out_vars
+        assert mb._entangled_qubits == set(entangled_qubits)
+
+    def test_cannot_add_duplicate_to_measure_block(self):
+        hw = get_default_echo_hardware()
+        targets = [hw.qubits[0], hw.qubits[-1]]
+        out_vars = ["c[0]", "b[1]"]
+        mb = MeasureBlock(
+            targets,
+            AcquireMode.INTEGRATOR,
+            output_variables=out_vars,
+        )
+        assert mb.quantum_targets == targets
+        with pytest.raises(ValueError):
+            mb.add_measurements(targets[1], AcquireMode.INTEGRATOR)
+
+    def test_measure_block_duration(self):
+        hw = get_default_echo_hardware()
+        target = hw.qubits[0]
+        mb = MeasureBlock([], AcquireMode.RAW)
+        assert mb.duration == 0.0
+        mb.add_measurements(target, AcquireMode.INTEGRATOR)
+        acq = mb.get_acquires(target)[0]
+        assert mb.duration > 0
+        assert mb.duration == pytest.approx(acq.delay + acq.duration)
+        mb._duration = 1
+        assert mb.duration == 1
+
+    def test_sequential_measure_block_from_builder(self):
+        """Test that measurements can be added in sequence."""
+        hw = get_default_echo_hardware()
+        builder = get_builder(hw)
+        assert len(builder._instructions) == 0
+        builder.measure(hw.get_qubit(0))
+        # Adding a non-post-processing Quantum instruction between measurements forces
+        # sequential measure blocks.
+        builder.synchronize(hw.qubits)
+        builder.measure(hw.get_qubit(2))
+        assert len(builder._instructions) == 3
+        assert builder._instructions[0].quantum_targets == [hw.get_qubit(0)]
+        assert builder._instructions[-1].quantum_targets == [hw.get_qubit(2)]
+
+    def test_fused_measure_block_from_builder(self):
+        """Test that measurements can be fused together."""
+        hw = get_default_echo_hardware()
+        builder = get_builder(hw)
+        assert len(builder._instructions) == 0
```

Comment:
private field access, isn't there a public getter ?

Reply:
The public getter expands the `InstructionBlock`s out to the full list, that complicates what I want to test here.